### DISCUSSION TITLE
Automated Real Estate Photo Enhancement using Deep Learning

### DISCUSSION THREAD

**Discussion by Rohit23333** [ultralytics/ultralytics#23716] [General]
**What I am trying to do:**
I am a real estate photographer. After every shoot I manually edit each photo in Lightroom/Photoshop to make them look professional — adjusting brightness, colour temperature, contrast, and white balance. This editing takes a lot of time for every property.
I want to build an AI model that learns my editing style from my past work and automatically applies the same edits to new raw photos — saving me hours of manual editing time per property.

**How it works:**
I have 426 pairs of photos — each pair has the original raw photo and my manually edited version. I want to train a model on these pairs so it learns the difference between my raw and edited photos and applies the same transformation to new raw photos automatically.

**Dataset:**
426 before/after image pairs
Before: Raw camera photos
After: Manually edited versions in Lightroom/Photoshop
Types: Bedrooms, kitchens, bathrooms, living rooms, exteriors

**Hardware:**
GPU: RTX 3070 Ti Laptop (8GB VRAM)
CPU: i7-12700H
RAM: 32GB
OS: Windows 11

**What I need help with:**
What is the best deep learning approach for this problem?
Is pix2pix the right model or is there a better alternative?
How can I get the output as close as possible to my manual edits?
Any suggestions for architecture, loss functions, or training strategy?
How to handle difficult cases like high contrast lighting and bright white rooms?

If you have worked on a similar problem or have any ideas on the best approach to achieve this, please share your thoughts and suggestions.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Rohit23333, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unblocked quickly—an Ultralytics engineer will also assist soon. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (your real-estate before/after enhancement project sounds like one), please provide as much information as possible, including dataset image examples (a few representative before/after pairs) and training logs (if you’ve already tried training anything), and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). If you can also share how your “raw” inputs are stored (true RAW vs exported JPEG/TIFF) and what resolution you plan to train/infer at, that context will help a lot 💡📸

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Ultralytics YOLO is for detection, so for “learn my Lightroom look from paired before/after” you’ll want a supervised image-to-image approach; `pix2pix` can work, but with 426 pairs you’ll usually get closer-to-authentic results by first learning a constrained global edit (WB/exposure/tone curve or a 3D LUT) from a consistent 16-bit export, then optionally adding a lightweight U-Net refinement trained with `L1` + perceptual (VGG) loss (use adversarial loss only if outputs look too smooth), and for harder lighting/white-room cases you can expand diversity via style/lighting augmentation like [Neural Style Transfer](https://www.ultralytics.com/glossary/neural-style-transfer)—are your edits mostly global, or do you use local masks/brushes (that detail changes the best architecture)?

**Comment by Rohit23333** [comment]
My edits are mostly global — I adjust brightness, exposure, colour temperature, white balance, contrast, and tone curves that affect the whole photo. I occasionally use local adjustments for windows (to reduce overexposure) and to brighten dark corners, but 80% of my editing is global adjustments applied to the entire image.

**Comment by Rohit23333** [comment]
Thanks for the detailed response. My edits are mostly global. Would you recommend adding a 3D LUT estimation network before the U-Net as a pre-processing step, or is it better to handle everything in one end-to-end network? Also, what specific VGG perceptual loss layers work best for colour/tone matching in interior photography?

**Reply by glenn-jocher** [reply]
If your edits are mostly global, I’d start with a LUT-first approach: train a 3D LUT estimation network (optionally end-to-end with a differentiable LUT layer) as the baseline, then only add a small residual/refinement U-Net if you see consistent failures on local effects like window highlights or mixed lighting; with just 426 pairs, the LUT path usually generalizes more cleanly than asking a big U-Net to learn everything at once. For perceptual loss, use shallow VGG features (e.g., `relu1_2` and `relu2_2`) and keep the weight modest—deeper layers tend to push semantics/texture more than colorimetry and can drift tone/color; the general VGG feature-extraction idea is the same one described in the [Neural Style Transfer (VGG feature layers)](https://www.ultralytics.com/glossary/neural-style-transfer) glossary entry.

**Comment by thanh99256** [comment]
Hi @Rohit23333, I am currently working on a very similar project for real estate photo editing. Did you end up having success with the 3D LUT + U-Net approach that Glenn suggested?

If you were successful, would you mind sharing a bit about your final methodology or workflow? Any tips or resources would be hugely appreciated. Thank you!

**Comment by nasy** [comment]
Hi, 

I am the founder of Stager AI, and after lots of hours we managed to train a model that performs automatic real estate HDR merging at high quality that can compare to fotello or autohdr. We tried everything I can ensure LUTs don't work as they don't adapt to each image conditions, pix2pix doesn't provide enough quality and resolution.. can't reveal the final recipe but it's not a single model it's a multi step process that's was quite complex to put together.

 https://stagerai.com/auto-hdr

**Reply by glenn-jocher** [reply]
Thanks for the added context—this use case is generally better treated as a multi-stage image-to-image pipeline than a single YOLO model, so for 426 paired raw→edited photos I’d start with patch-based U-Net/ResNet or `pix2pixHD`-style training using `L1 + SSIM + perceptual loss`, strict pair alignment plus input normalization as outlined in this [self-supervised denoising breakdown](https://www.ultralytics.com/blog/self-supervised-learning-for-denoising-a-step-by-step-breakdown), and a separate HDR/exposure-fusion step for windows and bright white rooms; if you share a few sample pairs and target resolution here, we can suggest a more concrete setup.

**Comment by breastonesthrowrealty-boop** [comment]
Hi, I’m new to this topic and really curious — could you explain more about the separate HDR step for windows? Why do windows need special treatment and how does it work in practice?

**Comment by thanh99256** [comment]
Hi Ignasi,

I’m new to this topic but really interested in what you’ve built. I’d love
to test your auto-HDR model with some real estate photos I have. Would it
be possible to try a few sample images through your tool at
stagerai.com/auto-hdr?

I’m particularly curious about how it handles window areas — that’s usually
the trickiest part in real estate photography.

Looking forward to hearing from you!

Vào Th 2, 15 thg 6, 2026 lúc 04:25 Ignasi Tuduri ***@***.***>
đã viết:

**Reply by glenn-jocher** [reply]
Please keep any sample tests and results in this discussion so others can benefit; for tricky window-heavy interiors, the strongest real-estate workflows are usually multi-stage with HDR/exposure fusion and local enhancement, not a single `pix2pix` pass.

**Comment by prestigegoldengrove386-bit** [comment]
Automated real estate photo enhancement using deep learning improves property images by optimizing lighting, clarity, and colors instantly. AI models enhance visual appeal, remove imperfections, and create realistic presentations. This helps attract buyers, increases engagement, and ensures faster marketing with professional-quality visuals, making listings more competitive in digital real estate platforms.
Visit site - https://hyderabadresidentialprojects.com/prestige/prestige-golden-grove/
